test/test_propagate_convolution.py

```diff
@@ -0,0 +1,151 @@
+"""Test PyDynamic.uncertainty.propagate_convolve"""
```

Comment:
These are awesomely careful chosen test cases. I would have loved to see hypothesis being facilitated to introduce the randomness, but these hand-crafted test cases are really covering all execution branches and you know best, if these touch corner cases as well. Things like the `random_array()` function are built into Hypothesis and come along with quite some intelligence for choosing the actual numerical values to discover erroneous corner cases.

Reply:
Yes, that is true. I should some day look much more into what hypothesis already has to offer. 

But the random valid covariance-matrix is probably a bit too specific 😄 (The convolution-algorithm itself should be quite robust against non-plausible covariance inputs, but in the Monte-Carlo comparison `np.random.multivariate_normal` requires something plausible.)